pkg/vsphere/adapter.go

```diff
@@ -69,13 +74,28 @@ func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClie
 		logger.Fatalf("Unable to determine source: %v", err)
 	}
 
+	kubeclient := ctx.Value(client.Key{})
+	if kubeclient == nil {
+		logger.Fatalf("NIL kubeclient")
+	}
+
+	// TODO: configmap name needs to be passed in the env, or the name of the source needs to
+	// plumbed in so we can derive it here.
+	// https://github.com/mattmoor/vmware-sources/issues/8
+	kvstore := kvstore.NewConfigMapKVStore(ctx, "test", env.Namespace, kubeclient.(*kubernetes.Clientset).CoreV1())
```

Comment:
What's with the type cast?

Reply:
Value returns interface{} if I don't cast it, I get this:
./adapter.go:85:79: kubeclient.CoreV1 undefined (type interface {} is interface with no methods)

🤷‍♂

Reply:
Oh, you are using `context.Value` not the injection stuff. 🤦‍♂ 

I'll add it to my shit list.

Reply:
actually you are using the key, so `kubeclient.Get(ctx)` should work here and it does the cast for you.

Reply:
LOL but you are using the key for this repo's client, not the kubernetes one 🤣